utils/polus-imagej-util/{{cookiecutter.project_slug}}/src/ij_converter.py

```diff
@@ -0,0 +1,250 @@
+        'int64'      : imglyb.types.LongType,
+        'uint8'      : imglyb.types.UnsignedByteType,
+        'uint16'     : imglyb.types.UnsignedShortType,
+        'uint32'     : imglyb.types.UnsignedIntType,
+        'uint64'     : imglyb.types.UnsignedLongType
+    }
+    # PRIMITIVES = {
+    #     'double'     : jpype.JDouble,
+    #     'float'      : jpype.JFloat,
+    #     'long'       : jpype.JLong,
+    #     'int'        : jpype.JInt,
+    #     'short'      : jpype.JShort,
+    #     'char'       : jpype.JChar,
+    #     'byte'       : jpype.JByte,
+    #     'boolean'    : jpype.JBoolean
+    # }
+    # PRIMITIVES = {
+    #     'double'     : jpype.JDouble,
+    #     'float'      : jpype.JFloat,
+    #     'long'       : jpype.JLong,
+    #     'int'        : jpype.JInt,
+    #     'short'      : jpype.JShort,
+    #     'char'       : jpype.JChar,
+    #     'byte'       : jpype.JByte,
+    #     'boolean'    : jpype.JBoolean
+    # }
+    PRIMITIVE_ARRAYS = {
+        'double[]'     : jpype.JDouble[:],
+        'float[]'      : jpype.JFloat[:],
+        'long[]'       : jpype.JLong[:],
+        'int[]'        : jpype.JInt[:],
+        'short[]'      : jpype.JShort[:],
+        'char[]'       : jpype.JChar[:],
+        'byte[]'       : jpype.JByte[:],
+        'boolean[]'    : jpype.JBoolean[:]
+    }
+    
+scyjava.when_jvm_starts(_java_setup)
+
+JAVA_CONVERT = {}
+# JAVA_CONVERT.update({
+#     t:lambda s,t,st: IMGLYB_PRIMITIVES[str(st)](st.type(s)) for t in ABSTRACT_SCALARS
+# })
+# JAVA_CONVERT.update({
+#     t:lambda s,t,st: PRIMITIVES[t](float(s)) for t in SCALARS
+# })
+# JAVA_CONVERT.update({
+#     t:lambda s,t,st: PRIMITIVE_ARRAYS[t]([float(si) for si in s.split(',')]) for t in ARRAYS
+# })
+JAVA_CONVERT.update({
+    t: lambda s,t,st: IMGLYB_PRIMITIVES[str(st)](s) for t in SCALARS
+})
+
+JAVA_CONVERT.update({
+    t:lambda s: imglyb.util.Views.iterable(ij.py.to_java(s)) for t in ABSTRACT_ITERABLES
+})
+JAVA_CONVERT.update({
+    t:lambda s: imglyb.util._to_imglib(s) for t in IMG_ARRAYS
+})
+
+def to_java(np_array,java_type,java_dtype=None):
+
+    if ij == None:
+        raise ValueError('No imagej instance found.')
+
+    if isinstance(np_array,type(None)):
+        return None
+
+    if java_type in JAVA_CONVERT.keys():
+        if str(java_dtype) != 'None':
+            out_array = JAVA_CONVERT[java_type](np_array,java_type,java_dtype)
+        else:
+            out_array = JAVA_CONVERT[java_type](np_array)
+    else:
+        logger.warning('Did not recognize type, {}, will pass default.'.format(java_type))
+        out_array = ij.py.to_java(np_array)
+        
+
+    return out_array
+
+def from_java(java_array,java_type):
+
+    if ij == None:
+        raise ValueError('No imagej instance found.')
+
+    if ij.py.dtype(java_array) == bool:
+        java_array = ij.op().convert().uint8(java_array)
+
+    return ij.py.from_java(java_array)
+
+if __name__ == '__main__':
```

Comment:
Everything under this looks good, but it belongs in some kind of unit test. Remember, that this is part of the cookiecutter template so all of this testing code is generated for each plugin.

I think a better solution is to add some unit tests into the `polus-imagej-util/tests` folder. Mainly because these are core tests, and if they pass in the core directory they should also pass for every plugin. I think these tests are also useful as pyimagej is still actively worked on, and it is important for us to be able to verify that our conversions work when/if we update the version of pyimagej we use.

Reply:
This has been moved to `polus-imagej-util/tests` folder however it has not been integrated with the `UnitTest` module.